src/controllers/auth.controller.js

```diff
@@ -0,0 +1,370 @@
+
+  checkUserExistence(foundUser, id);
+
+  if (!updatedName) {
+    throw ApiError.badRequest('You need to provide an updated name!');
+  }
+
+  if (updatedName === foundUser.name) {
+    throw ApiError.badRequest('No need to change your name!', {
+      name: 'Your updated name is the same as the old one!',
+    });
+  }
+
+  await userService.updateName(foundUser, updatedName);
+
+  res.status(200).send(successfulMessage(foundUser, 'name'));
+};
+
+const updatePassword = async(req, res) => {
+  const { id } = req.params;
+  const { oldPassword, newPassword, confirmation } = req.body;
+
+  if (!oldPassword || !newPassword || !confirmation) {
+    throw ApiError.badRequest(
+      'You need to provide old password, new password and confirmation!'
+    );
+  }
+
+  const foundUser = await userService.getUserById(id);
+
+  checkUserExistence(foundUser, id);
+
+  if (!(await bcrypt.compare(oldPassword, foundUser.password))) {
+    throw ApiError.badRequest('Old Password is wrong!');
+  }
+
+  if (newPassword === oldPassword) {
+    throw ApiError.badRequest(
+      'New password should be different from your old one!'
+    );
+  }
+
+  const passwordValidationError = validatePassword(newPassword);
+
+  if (passwordValidationError) {
+    throw ApiError.badRequest('Validation error', {
+      newPassword: passwordValidationError,
+    });
+  }
+
+  if (newPassword !== confirmation) {
+    throw ApiError.badRequest('The new password and confirmation do not match');
+  }
+
+  await userService.updatePassword(foundUser, newPassword);
+
+  res.status(200).send(successfulMessage(foundUser, 'password'));
+};
+
+const sendEmailConfirmation = async(req, res) => {
+  const { id } = req.params;
+  const { email, password } = req.body;
+
+  if (!email || !password) {
+    throw ApiError.badRequest(
+      'You need to provide a new email and current password!'
+    );
+  }
+
+  const foundUser = await userService.getUserById(id);
+
+  checkUserExistence(foundUser, id);
+
+  if (!(await bcrypt.compare(password, foundUser.password))) {
+    throw ApiError.badRequest('Incorrect password for changing email!');
+  }
+
+  if (email === foundUser.email) {
+    throw ApiError.badRequest(
+      'Your new email should be different from your current one!'
+    );
+  }
+
+  await userService.sendEmailConfirmation(foundUser, email, res);
+
+  res.status(200).send({
+    message: `Confirm your new email please!
+        You have been sent an email with confirmation!`,
+  });
+};
+
+const updateEmail = async(req, res) => {
+  const { confirmationToken } = req.params;
+
+  if (!confirmationToken) {
+    throw ApiError.badRequest('Missing confirmation token!');
+  }
+
+  const foundUser = await userService.updateEmail(confirmationToken);
+
+  res.clearCookie('confirmationToken');
+
+  res.status(200).send({
+    message: 'Your email has been updated successfully!',
+    updatedUser: foundUser,
+  });
+};
+
+const authorizeWithGoogle = (req, res) => {
```

Comment:
is this a fake google auth, or is it a mistake? because you take accessToken from request and sending the same on in response